libtransmission/crypto-utils.cc

```diff
@@ -106,15 +106,15 @@ int tr_rand_int_weak(int upper_bound)
 {
     TR_ASSERT(upper_bound > 0);
 
-    static bool init = false;
-
-    if (!init)
+    static std::mt19937 random_engine = []()
     {
-        srand(tr_time_msec());
-        init = true;
-    }
+        std::random_device random_device;
+        return std::mt19937{ random_device() };
+    }();
+    static std::uniform_int_distribution<> distribution{ 0 };
```

Comment:
`0` is the default, and you're always passing lower bound below anyway, so explicitly specifying it here makes little sense.
```suggestion
    static auto distribution = std::uniform_int_distribution<>{};
```